homeassistant/components/light/template.py

```diff
@@ -0,0 +1,192 @@
+            self._template.hass = self.hass
+        if self._level_template is not None:
+            self._level_template.hass = self.hass
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        if self._level_script is not None:
+            return SUPPORT_BRIGHTNESS
+
+        return 0
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        state = yield from async_get_last_state(self.hass, self.entity_id)
+        if state:
+            self._state = state.state == STATE_ON
+
+        @callback
+        def template_light_state_listener(entity, old_state, new_state):
+            """Handle target device state changes."""
+            self.hass.async_add_job(self.async_update_ha_state(True))
+
+        @callback
+        def template_light_startup(event):
+            """Update template on startup."""
+            async_track_state_change(
+                self.hass, self._entities, template_light_state_listener)
+
+            self.hass.async_add_job(self.async_update_ha_state(True))
+
+        self.hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_START, template_light_startup)
+
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
+        """Turn the light on."""
+        self._state = True
+        if ATTR_BRIGHTNESS in kwargs and self._level_script:
+            self._brightness = kwargs[ATTR_BRIGHTNESS]
+            self.hass.async_add_job(self._level_script.async_run(
+                {"brightness": kwargs[ATTR_BRIGHTNESS]}))
+        else:
+            self.hass.async_add_job(self._on_script.async_run())
+
+    @asyncio.coroutine
+    def async_turn_off(self, **kwargs):
+        """Turn the light off."""
+        self.hass.async_add_job(self._off_script.async_run())
+        self._state = False
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Update the state from the template."""
+        try:
+            if self._template is not None:
+                state = self._template.async_render().lower()
+
+                if state in _VALID_STATES:
+                    self._state = state in ('true', STATE_ON)
+                else:
+                    _LOGGER.error(
+                        'Received invalid light is_on state: %s. ' +
+                        'Expected: %s',
+                        state, ', '.join(_VALID_STATES))
+                    self._state = None
+
+            if self._level_template is not None:
+                brightness = self._level_template.async_render()
+                self._brightness = brightness
+
+        except TemplateError as ex:
+            _LOGGER.error(ex)
+            self._state = None
```

Comment:
Since there are two renders here, wrap each one separately, and set the appropriate state to `None`. It's safe to have this try only wrap the lines with `async_render()`.